hw/ip/usbdev/doc/_index.md

```diff
@@ -66,54 +66,79 @@ synchronized across the clock domain boundary. A dual-port
 asynchronous buffer SRAM is used for data transfers between the bus
 clock and USB clock domains.
 
-## USB Interface Pins
-
-The interface pin table summarizes the external pins.
 
+## USB Interface Pins
 
-|External Pin|Internal Signal|Notes|
-|------------|---------------|-----|
-|USB D+, USB D-|usb_d_i, usb_dp_i, usb_dn_i, usb_d_o, usb_se0_o|Interface to a differential USB transceiver. This interface can be selected by writing 1 to {{< regref "phy_config.rx_differential_mode" >}} and {{< regref "phy_config.tx_differential_mode" >}}. In differential mode, the single-ended signals usb_dp_i and usb_dn_i are used to detect the SE0 state.|
-|USB D+, USB D-|usb_dp_i, usb_dn_i, usb_dp_o, usb_dn_o|Single-ended interface to regular IO cells. This interface can be used for prototyping on an FPGA, but will probably not be USB compliant. This interface can be selected by writing 0 to {{< regref "phy_config.rx_differential_mode" >}} and {{< regref "phy_config.tx_differential_mode" >}}.|
-||usb_oe_o|Enable driving the external pins.|
-|[usb_pullup]|usb_pullup_o|When the usb_pullup_o asserts a 1.5k pullup resistor should be connected to D+. This can be done inside the chip or with an external pin. A permanently connected resistor can also be used.|
-|[TX Mode]|tx_mode_se_o|Indicates the selected TX mode. 1 corresponds to single-ended operation.|
-|usb_sense|usb_sense_i|The sense pin indicates the presence of VBUS from the host.|
+Full-Speed USB uses a bidirectional serial interface as shown in Figure 7-24 of the [USB 2.0 Full-Speed specification](https://www.usb.org/document-library/usb-20-specification).
+For reasons of flexibilty, this IP block features both differential and single-ended transmit and receive paths.
 
-The USB 2.0 Full Speed specification uses a bidirectional serial interface that can be implemented with pseudo-differential 3.3V GPIO pins and an oversampling receiver for recovery of the bitstream and clock alignment.
-The IP interface for the D+ and D- pins is presented using a pair of transmit signals, a pair of receive signals and a transmit enable.
-External to the IP these should be combined to drive the pins when transmit is enabled and receive when transmit is not enabled.
+For better receive sensitivity, lower transmit jitter and to be standard compliant, the differential paths should be used together with a dedicated USB PHY such as the USB1T11A or the USB1T20.
```

Comment:
Done.

Reply:
I originally had _must_ there. I don't think it's possible to be standard compliant without a differential receiver. c.f. Section 7.1.4.1 of the spec:

> A differential input receiver must be used to accept the USB data signal. The receiver must feature an input sensitivity (V DI ) of at least 200 mV when both differential data inputs are in the differential common mode range (V CM ) of 0.8 V to 2.5 V, as shown in Figure 7-19.

I don't see a way in which this could be done without a differential receiver. Similarly difficult requirements exist for the transmitter: Output impedance, crossover point and transition slope. I don't think those can be met with standard IO pads either.

You can of course still use standard IOs on your chip, but I think that the public documentation should be geared towards standard compliance.

Reply:
Thanks @stefanlippuner for your input. I think we are mixing up the "terms" differential receiver and differential receive path here. It seems right that to be standard compliant, a differential receiver must be used.

However, looking at the datasheets of the two referenced differential transceivers, I see that both feature a RCV output (corresponds to our `d_i`) as well the V_P and V_M (the gated D+ and D- corresponding to our `dp_i` and `dn_i`). I think Mark means that in the test chip, we could use those `dp_i` and `dn_i` without `d_i`. This corresponds to using the single-ended receive path in usbdev in combination with a differential transceiver.

I will adapt the documentation to make this clear.